### PR TITLE
Tiered Storage Improvements

### PR DESCRIPTION
**Pull request by gene-redpanda**
Contains testing changes to incorporate tiered storage including an S3 bucket check,  a tiered storage playbook and fixes for template merging. Also incorporates some changes to the docker container to support running awscli.

### CODE REVIEW

**Comment by gene-redpanda** [comment]
Screenshot of populated bucket to show it is working:

Screenshot 2023-05-11 at 1 26 06 PM

File: aws/s3.tf [hunk 1]
```diff
@@ -5,11 +5,20 @@ resource "aws_s3_bucket" "tiered_storage" {
...
+#  acl    = "private"
+#}
+#
+#resource "aws_s3_bucket_ownership_controls" "s3_bucket_acl_ownership" {
+#  for_each = toset(aws_s3_bucket.tiered_storage[*].id)
+#  bucket   = each.key
+#  rule {
+#    object_ownership = "ObjectWriter"
+#  }
+#}
```
Comment (r-vasquez) [inline aws/s3.tf:21]: [question] Is this intended?

Comment (gene-redpanda) [reply aws/s3.tf:21]: Yes, the commented docs explain why in the warning doc. Basically ACL + IAM == error and not recommended.

**Comment by twmb** [comment]
Can we delete the code rather than keep it commented?

**Review by r-vasquez — APPROVED** [review]

**Review by twmb — CHANGES_REQUESTED** [review]
commit is a bit odd now, can it be renamed to something like
`adds tiered storage playbook & testing, improves container to support awscli`
?

**Comment by gene-redpanda** [comment]
> commit is a bit odd now, can it be renamed to something like `adds tiered storage playbook & testing, improves container to support awscli` ?

Done!

**Review by twmb — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -232,13 +224,17 @@ See: https://stackoverflow.com/questions/50168647/multiprocessing-causes-python-
...
-To test with a specific commit of RAC, set the following value in your environment. 
+To test with a specific commit of RAC, set the following value in your environment.
+
 ```shell
 export ANSIBLE_COLLECTION_INSTALL=git+https://github.com/redpanda-data/redpanda-ansible-collection.git,<<YOUR COMMIT HERE>>
 ```
-The alphanumeric string after the comma should be the short form of the commit you want installed. This value is consumed by the task file as part of an ansible-galaxy install command.
 
-More documentation is available [here](https://docs.ansible.com/ansible/latest/collections_guide/collections_installing.html)
+The alphanumeric string after the comma should be the short form of the commit you want installed. This value is
```
Comment (hcoyote) [inline README.md:233]: commit meaning the git commit hash?

File: ansible/provision-tiered-storage-cluster.yml [hunk 3]
```diff
@@ -0,0 +1,32 @@
+# Creates a TLS enabled cluster with tiered storage local CA in artifacts/ca for testing purposes
```
Comment (hcoyote) [inline ansible/provision-tiered-storage-cluster.yml:1]: minor:

Think this is missing a word or two. maybe this?

"with tiered storage, using a local CA [...]"

File: aws/cluster.tf [hunk 4]
```diff
@@ -31,7 +31,8 @@ resource "aws_iam_policy" "redpanda" {
           "s3-object-lambda:*",
         ],
         "Resource" : [
-          "arn:aws:s3:::${local.tiered_storage_bucket_name}/*"
+          "arn:aws:s3:::${local.tiered_storage_bucket_name}/*",
+          "arn:aws:s3:::${local.tiered_storage_bucket_name}"
```
[the remark this answers is not in the document]
Comment (gene-redpanda) [reply aws/cluster.tf:35]: it was, just squashed in the merge commit to get it down to to the desired single commit.

**Review by hcoyote — APPROVED** [review]

**Pull request merged**